mlflow/utils/name_utils.py

```diff
@@ -0,0 +1,320 @@
+import random
+
+
+def _generate_string(sep, integer_scale):
+
+    predicate = random.choice(_GENERATOR_PREDICATES).lower()
+    noun = random.choice(_GENERATOR_NOUNS).lower()
+    num = random.randint(0, 10**integer_scale)
+    return f"{predicate}{sep}{noun}{sep}{num}"
+
+
+def _generate_random_name(sep="-", integer_scale=3, max_length=20):
+    """Helper function for generating a random predicate, noun, and integer combination
+
+    :param sep: String seperator for word spacing
+    :param integer_scale: dictates the maximum scale range for random integer sampling (power of 10)
+    :param max_length: maximum allowable string length
+
+    :return: A random string phrase comprised of a predicate, noun, and random integer
+    """
+    max_iter = 10
+    i = 0
+    while True:
+        name = _generate_string(sep, integer_scale)
+        if len(name) <= max_length:
+            return name
+        elif i == max_iter:
+            return name[:max_length]
+        else:
+            i += 1
```

Comment:
```suggestion
    for _ in range(10):
        name = _generate_string(sep, integer_scale)
        if len(name) <= max_length:
            return name
    return name[:max_length]
```

nit: just for simplifying the code.

Reply:
much better :)